mixer/test/spyAdapter/template/template.gen.go

```diff
@@ -423,13 +425,14 @@ var (
 			},
 
 			// DispatchCheck dispatches the instance to the handler.
-			DispatchCheck: func(ctx context.Context, handler adapter.Handler, inst interface{}) (adapter.CheckResult, error) {
+			DispatchCheck: func(ctx context.Context, handler adapter.Handler, inst interface{}) (adapter.CheckResult, interface{}, error) {
```

Comment:
The reason that the type is `interface{}` instead of the more precise `Output` is because this function signature has to be generic across all adapters. Without generics, we have to fall for the least common denominator, and then type-case within the function bodies.

Note that things will become better with OOP adapters. We should be able to generate a type-safe remote procedure call instead of type-casting within the process.